lib/_emerge/depgraph.py

```diff
@@ -1235,13 +1235,13 @@ def _show_ignored_binaries_respect_use(self, respect_use):
         seen = {}
         messages = []
         merging = {
-            pkg.cpv
+            (pkg.root, pkg.cpv)
             for pkg in self._dynamic_config._displayed_list
             if isinstance(pkg, Package)
         }
         for pkg, flags in respect_use.items():
-            # Don't include recursive deps which aren't in the merge list anyway.
-            if pkg.cpv not in merging:
+            # Don't include recursive deps wohich aren't in the merge list anyway.
```

Comment:
Huh. Wonder how that happened. :D

Reply:
(I've just realised I need to fix `seen` for root as well)